src/System.Data.SqlClient/src/System/Data/SqlClient/SqlConnection.cs

```diff
@@ -748,6 +764,19 @@ private void CancelOpenAndWait()
         public override Task OpenAsync(CancellationToken cancellationToken)
         {
             PrepareStatisticsForNewConnection();
+            
+            if (StatisticsEnabled ||
+                _diagnosticListener.IsEnabled(SqlClientDiagnosticListenerExtensions.SqlAfterExecuteCommand))
+            {
+                if (null == _statistics)
```

Comment:
The PrepareStatisticsForNewConnection method (called above) initializes the _statistics as well. Why is _statistics initialization needed again?